extensions/2.0/Khronos/KHR_xmp/README.md

```diff
@@ -0,0 +1,337 @@
+
+An indirection level is introduced to avoid replicating the same XMP metadata in multiple glTF nodes.
+XMP metadatas are defined within a dictionary property in the glTF scene description file by adding an `extensions` property to the top-level glTF 2.0 object and defining a `KHR_xmp` object. The `KHR_xmp` object defines two properties:
+
+* `namespaces` : a dictionary mapping XMP [namespaces](https://github.com/adobe/xmp-docs/blob/master/Namespaces.md) to the URI where they are defined.
+* `packets`: an array of metadata packets referencing the namespaces defined in the `namespaces` dictionary.
+
+The following example defines a glTF scene with a sample XMP metadata.
+
+```json
+"extensions": {
+    "KHR_xmp" : {
+        "namespaces" : {
+            "dc" : "http://purl.org/dc/elements/1.1/"
+        },
+        "packets" : [
+            {
+                "dc:contributor" : [ "Creator1Name", "Creator2Email@email.com", "Creator3Name<Email@email.com>"],
+                "dc:coverage" : "Bay Area, California, United States",
+                "dc:creator" : [ "CreatorName", "CreatorEmail@email.com"],
+                "dc:date" : ["1997-07-16T19:20:30+01:00"],
+                "dc:description" : { "en-us": "text"},
+                "dc:format" : "model/gltf-binary",
+                "dc:identifier" : "urn:stock-id:292930",
+                "dc:language" : ["en"],
+                "dc:publisher" : ["Company"],
+                "dc:relation" : ["https://www.khronos.org/"],
+                "dc:rights" : {"en-us": "BSD"},
+                "dc:source" : "http://related_resource.org/derived_from_this.gltf",
+                "dc:subject" : ["architecture"],
+                "dc:title" : { "en-us" : "MyModel", "it-IT" : "Mio Modello"},
+                "dc:type" : ["Physical Object"],
+            }
+        ]
+    }
+}
+```
+
+## Instantiating XMP metadata
+
+Metadata can be instantiated in a node by defining the `extensions.KHR_xmp` property and, within that, an index into the `packets` array using the `packet` property.
+The following example shows a glTF Mesh instantiating the XMP metadata at index `0`.
+
+```json
+{
+    "meshes": [
+      {
+        "extensions" : {
+            "KHR_xmp": {
+                "packet" : 0,
+            }
+        },
+        "primitives": [
+          {
+            "attributes": {
+                "NORMAL": 23,
+                "POSITION": 22,
+                "TANGENT": 24,
+                "TEXCOORD_0": 25
+            },
+            "indices": 21,
+            "material": 3,
+            "mode": 4
+          }
+        ]
+      }
+    ]
+}
+```
+
+### Notes:
+
+#### Overriding Rule
+
+A glTF scene might reference resources already containing XMP metadata.
+A relevant example would be an image node referencing a PNG file with embedded XMP metadata.
+When this happens the XMP metadata specified in the glTF manifest takes precedence over the metadata embedded in the referenced resource.
```

Comment:
It's not about overriding but more of inheritance.  If someone sets a `dc:author` on a node, then they should be considered the author of everything on that node *that is not otherwise specified*.  So in the example you gave, if a given image/PNG had embedded XMP with a different `dc:author` that applies for that specific asset.

Reply:
Your phrasing of precedence over everything "that is not otherwise specified" sounds like the exact opposite of the phrasing currently in this PR, "XMP metadata specified in the glTF manifest takes precedence over the metadata embedded in the referenced resource" to me — I think the PR should be reworded, if so?

Reply:
Yes, the PR needs to be reworded.

Reply:
Taking your example Don, I was try to resolve the situation in which we have, for instance, an image object with KHR_xmp metadata that references a `png` with metadata. Which metadata should take precedence?
```
"images": [{
  "uri": "duckCM.png", 
  "extensions" : { "KHR_xmp": { "packet" : 0 } },
}]
```
My rationale was to give the option to the container (glTF) to override the referenced metadata.  I have no strong opinion though, I  just wanted to make sure this situation is addressed. It sounds like there is consensus in doing the opposite -> I am happy to make the change.

Reply:
Supporting XMP metadata embedded in PNG files would require:
- parsing PNG chunks;
- decoding zlib/deflate compression;
- parsing XML-based XMP.

Do we expect web-based viewers to implement all of these?

Reply:
Actually, it doesn't.  XMP is stored as plain-text in PNG (and pretty much every other file format), so you can do a simple regex (or equiv) search to find the XMP - but then yes, you'd need to parse the XML+RDF.

Reply:
Ah, XMP specifically disallows deflate on its chunks, interesting. 

Still, would it be acceptable for glTF viewers to not implement embedded XMP support (PNG and JPEG) with this new spec update? So far, glTF web loaders just pass such images directly to the web browser without any processing.

Reply:
I would think that is up to each processor @lexaknyazev , depending on what they are trying to achieve in their processing of the glTF